cachito/web/models.py

```diff
@@ -308,7 +308,7 @@ def to_top_level_purl(self, request):
         """
         if self.type in ("gomod", "go-package"):
             return self.to_purl()
-        elif self.type in ("npm", "pip"):
+        elif self.type in ("npm", "pip", "git-submodule"):
             return self.to_vcs_purl(request.repo, request.ref)
```

Comment:
IMO there are 2 ways to identify a git submodule using a purl (in this example, the "tour" submodule in https://github.com/nirzari/retrodep):
1. main repo + submodule path: `pkg:github/nirzari/retrodep@abfebf169e8d67fcf5de07d41bbe9b1ea132491b#tour`
2. submodule repo: `pkg:github/nirzari/tour@522fb816eec295ad58bc488c74b2b46748d471b2`

This is option 1, except the subpath is missing (I'm working on that)
As far as I can tell, the only immediate benefit of option 2 would  be that it is correct even without subpath support, so option 1 is probably fine. I could be missing something though, cc @athos-ribeiro

Reply:
Since we've agreed on option 2 offline, here's some more details:
- the `to_purl()` method should be updated to handle "git-submodule" packages
    - you can use the `to_vcs_purl()` method, but you should parse the `repo_url` and `ref` from `self.version` (as stored in the DB)
    - see the handling for pip vcs dependencies https://github.com/release-engineering/cachito/blob/b61783d1590a220570ad3f1643e4a46f83360b3d/cachito/web/models.py#L246-L251
- in `to_top_level_purl()`, "git-submodule" packages should just call `to_purl()` (like "gomod" and "go-package")